pkgs/top-level/all-packages.nix

```diff
@@ -17523,9 +17523,11 @@ in
     inherit (darwin.apple_sdk.frameworks) CoreServices AudioUnit Cocoa;
   };
 
-  pulseaudio = callPackage ../servers/pulseaudio {
+  pulseaudio = callPackage ../servers/pulseaudio ({
     inherit (darwin.apple_sdk.frameworks) CoreServices AudioUnit Cocoa;
-  };
+  } // stdenv.lib.optionalAttrs stdenv.isDarwin {
+    autoreconfHook = buildPackages.autoreconfHook269;
```

Comment:
Why is this conditional on darwin only?

Reply:
It seems to be building fine on linux, so this avoids extra rebuild in staging-next

Reply:
Humm, then we should probably still make it unconditional in a followup on staging.

Can you post the build failure log here?

Reply:
Hmm, yeah I do usually try to keep linux and darwin versions aligned, but in this case wouldn't we want to use the default autoconf when it works fine? What do you think?

Error log: https://hydra.nixos.org/build/133894785/nixlog/1

Reply:
Yeah, maybe just add a

> with default autoreconfHook fails on darwin with "configure: error: *** Compiler does not support -std=gnu11" 

comment?

Reply:
The clang used here does not support c(++?)11 with gnu extensions. This is either a bug in autoconf or the used compiler changes.

Reply:
Comment added. Agree, I don't see any reason this shouldn't work on macOS, so possibly an autoconf bug. I opened an issue with pulseaudio in case it's a problem with their macros: https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/issues/1084